File: tests/Web.AcceptanceTests/Pages/SettingsPage.cs
namespace LiftAndShift.Web.AcceptanceTests.Pages;

public class SettingsPage(IPage page) : BasePage(page)
{
    public override string PagePath => $"{BaseUrl}/settings";

    public Task AssertHeading() => Assertions.Expect(Page.Locator("h1")).ToHaveTextAsync("Settings");

    public Task SelectLbs() => Page.Locator("input[type='radio'][value='lbs']").CheckAsync();

    public Task SelectKg() => Page.Locator("input[type='radio'][value='kg']").CheckAsync();

    public Task<bool> IsLbsChecked() => Page.Locator("input[type='radio'][value='lbs']").IsCheckedAsync();

    public Task ClickSaveSettings() => Page.Locator("button:has-text('Save Settings')").ClickAsync();

    public Task AssertSavedMessageVisible() => Assertions.Expect(Page.Locator("text=Settings saved.")).ToBeVisibleAsync();
}

File: src/Web/ClientApp/e2e/pages/settings.page.ts
import { expect } from '@playwright/test';
import { BasePage, baseUrl } from './base-page';

export class SettingsPage extends BasePage {
  get pagePath(): string {
    return `${baseUrl()}/settings`;
  }

  assertHeading() {
    return expect(this.page.locator('h1')).toHaveText('Settings');
  }

  selectLbs() {
    return this.page.locator("input[type='radio'][value='lbs']").check();
  }

  selectKg() {
    return this.page.locator("input[type='radio'][value='kg']").check();
  }

  isLbsChecked() {
    return this.page.locator("input[type='radio'][value='lbs']").isChecked();
  }

  clickSaveSettings() {
    return this.page.locator("button:has-text('Save Settings')").click();
  }

  assertSavedMessageVisible() {
    return expect(this.page.locator('text=Settings saved.')).toBeVisible();
  }
}
